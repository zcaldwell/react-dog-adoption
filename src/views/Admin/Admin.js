import React from 'react';
import DogForm from '../../components/DogForm';
import { useState } from 'react';
import { addDog } from '../../services/dogs';
import { useHistory } from 'react-router-dom';

export default function Admin() {
  const [dog, setDog] = useState({});
  const history = useHistory();

  const updateDogForm = (key, value) => {
    dog[key] = value;
    setDog({ ...dog });
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      await addDog(dog);
      alert('Success!');
      history.push(`/Dogs/`);
    } catch {
      alert('failed');
    }
  };

  return (
    <div>
      <DogForm {...dog} handleSubmit={handleSubmit} updateDogForm={updateDogForm} />
    </div>
  );
}
