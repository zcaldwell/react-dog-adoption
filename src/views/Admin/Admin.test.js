import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Admin from './Admin';

test('Dogs should display all dogs', async () => {
  const { container } = render(
    <MemoryRouter>
      <Admin />
    </MemoryRouter>
  );
  await screen.findByText('Name');
  expect(container).toMatchSnapshot();
});
