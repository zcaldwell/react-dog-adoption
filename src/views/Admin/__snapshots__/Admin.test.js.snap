// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dogs should display all dogs 1`] = `
<div>
  <div>
    <div
      class="editor"
    >
      <form>
        <div
          class="form-control"
        >
          <label>
            Name
          </label>
          <input
            placeholder="name"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-control"
        >
          <label>
            Age
          </label>
          <input
            placeholder="age"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-control"
        >
          <label>
            Image
          </label>
          <input
            placeholder="image"
            type="text"
            value=""
          />
          <div
            class="form-control"
          >
            <label>
              Bio
            </label>
            <textarea
              placeholder="bio"
            />
          </div>
          <div
            class="form-control"
          >
            <label>
              Breed
            </label>
            <input
              placeholder="breed"
              type="text"
              value=""
            />
          </div>
          <div
            class="submit-button"
          >
            <button>
              save
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
